src/ControlSystem/ControlErrors/Expansion.hpp

```diff
@@ -80,19 +83,32 @@ struct Expansion : tt::ConformsTo<protocols::ControlError> {
     using center_A = control_system::QueueTags::Center<::ah::ObjectLabel::A>;
     using center_B = control_system::QueueTags::Center<::ah::ObjectLabel::B>;
 
+    ASSERT(domain.excision_spheres().count(
+               "PrimaryRightObjectAExcisionSphere") == 1,
+           "Excision sphere for ObjectA not in the domain but is needed to "
+           "compute Expansion control error.");
+    ASSERT(domain.excision_spheres().count(
+               "SecondaryLeftObjectBExcisionSphere") == 1,
+           "Excision sphere for ObjectB not in the domain but is needed to "
+           "compute Expansion control error.");
+
     const double grid_position_of_A =
-        domain.excision_spheres().at("ObjectAExcisionSphere").center()[0];
+        domain.excision_spheres()
+            .at("PrimaryRightObjectAExcisionSphere")
```

Comment:
@nilsdeppe and I came up with a new suggestion:

- Since the domain creators and the control systems care about the two excisions being "left" or "right" we use that for option names in the domain creator (quite intuitive when writing input files) and for control system things in the code. This deep into the code we can assume people know about the grid frame which makes "left" and "right" unambiguous.
- For output datasets we let the user choose subfile names for the two horizons. We should let the user choose subfile names anyway so they know where to look for the output data just by reading the input file. Of course this means we offload the problem of naming the horizons to the user, but they can choose what makes most sense for their simulation and anyone can use the input file to get the names. Probably "Primary" and "Secondary" are good names here for BBH simulations, but for a comparison run with SpEC the user could name them "A" and "B" instead. This also means that output files from different simulations can have different subfile names, which is always the case when we let users choose subfile names (or file names) in the input file. Since the names are all in the input file, I think this is better than hardcoded names that magically appear in the output files. For archiving runs / building catalogs we can consider normalizing the names later, using the input files. Note that input files are also written into the H5 files and can be extracted with `spectre extract-input`. The input files could look like this:

```yaml
DomainCreator:
  BinaryCompactObject:
    ObjectRight:  # Currently "ObjectA"
      XCoord: 8.
      # ...

ApparentHorizons:
  Right:  # Currently "AhA"
    SubfileName: "Primary"  # New option
    FastFlow:
      # ...
```